src/test/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMBuilderTest.java

```diff
@@ -741,8 +741,8 @@ public void given__cloud_branch_norev_userpass__when__build__then__scmBuilt() th
     public void given__cloud_branch_norev_userkey__when__build__then__scmBuilt() throws Exception {
         createGitHubSCMSourceForTest(false, null);
         BranchSCMHead head = new BranchSCMHead("test-branch");
-        source.setCredentialsId("user-key");
         GitHubSCMBuilder instance = new GitHubSCMBuilder(source, head, null);
+        instance.withCredentials("user-key", GitHubSCMBuilder.SSH);
```

Comment:
Nit: prefer non deprecated API usage
```suggestion
        GitHubSCMBuilder instance = new GitHubSCMBuilder(source, head, revision)
                .withCredentials("user-key")
                .withResolver(GitHubSCMBuilder.SSH);
```